test/libsolidity/syntaxTests/functionCalls/new_with_invalid_calloptions.sol

```diff
@@ -14,14 +14,14 @@ contract C {
 // ====
 // EVMVersion: >=constantinople
 // ----
-// TypeError: (64-98): Cannot set option "value" on a non-payable function type.
+// TypeError: (64-98): Constructor for contract contract D must be payable for member "value" to be available.
 // TypeError: (64-98): Function call option "gas" cannot be used with "new".
 // TypeError: (102-123): Unknown call option "slt". Valid options are "salt", "value" and "gas".
-// TypeError: (102-123): Cannot set option "value" on a non-payable function type.
+// TypeError: (102-123): Constructor for contract contract D must be payable for member "value" to be available.
 // TypeError: (127-139): Unknown call option "val". Valid options are "salt", "value" and "gas".
 // TypeError: (143-172): Duplicate option "salt".
-// TypeError: (176-199): Cannot set option "value" on a non-payable function type.
-// TypeError: (176-199): Cannot set option "value" on a non-payable function type.
+// TypeError: (176-199): Constructor for contract contract D must be payable for member "value" to be available.
```

Comment:
Or maybe:
Constructor for contract contract D must be payable for option "value" to be available.

Reply:
Why "contract" twice? Since it's part of the type name, you could just not prepend "contract" in the error message.

Reply:
And yeah - maybe similar to the other case:
``Cannot set option "value", since the constructor of contract D is not payable.``

Reply:
Updated